### PR TITLE
fix: register consensus params

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

After upgrade:

```
~ % grpcurl \
    -plaintext \
    localhost:55894 \
    cosmos.consensus.v1.Query/Params
{
  "params": {
    "block": {
      "maxBytes": "5000000",
      "maxGas": "300000000"
    },
    "evidence": {
      "maxAgeNumBlocks": "100000",
      "maxAgeDuration": "172800s",
      "maxBytes": "1048576"
    },
    "validator": {
      "pubKeyTypes": [
        "ed25519"
      ]
    },
    "version": {

    }
  }
}
```

## Testing and Verifying

Queried for the value pre v22 upgrade on E2E, does not exist.

Queried for the value post v22 upgrade on E2E, exists and is upgraded to the above values.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: app/upgrades/v22/upgrades.go [hunk 1]
```diff
@@ -23,6 +25,13 @@ func CreateUpgradeHandler(
 			return nil, err
 		}
 
+		// Properly register consensus params. In the process, change params as per:
+		// https://forum.osmosis.zone/t/raise-maximum-gas-to-300m-and-lower-max-bytes-to-5mb/1116
+		defaultConsensusParams := tmtypes.DefaultConsensusParams().ToProto()
+		defaultConsensusParams.Block.MaxBytes = 5000000
+		defaultConsensusParams.Block.MaxGas = 300000000
+		keepers.ConsensusParamsKeeper.Set(ctx, &defaultConsensusParams)
```
Comment (czarcas7ic) [inline app/upgrades/v22/upgrades.go:33]: Need an ack that we are good with all defaults except for the MaxBytes and MaxGas here

File: tests/e2e/configurer/config/constants.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ const (
 	// number of blocks it takes to vote for a single validator to vote for a proposal
 	PropVoteBlocks float32 = 1
 	// number of blocks used as a calculation buffer
-	PropBufferBlocks float32 = 8
+	PropBufferBlocks float32 = 30
```
Comment (czarcas7ic) [inline tests/e2e/configurer/config/constants.go:17]: In the previous version of E2E, the block speed pre upgrade was normal, and the block speed post upgrade was accelerated. Now, block speed is accelerated both pre and post upgrade, so I increased the buffer accordingly.

File: app/upgrades/v22/upgrades.go [hunk 3]
```diff
@@ -23,6 +25,13 @@ func CreateUpgradeHandler(
 			return nil, err
 		}
 
+		// Properly register consensus params. In the process, change params as per:
+		// https://forum.osmosis.zone/t/raise-maximum-gas-to-300m-and-lower-max-bytes-to-5mb/1116
+		defaultConsensusParams := tmtypes.DefaultConsensusParams().ToProto()
+		defaultConsensusParams.Block.MaxBytes = 5000000
+		defaultConsensusParams.Block.MaxGas = 300000000
```
Comment (mattverse) [inline app/upgrades/v22/upgrades.go:32]: just for context, what's the reason we're only dealing with max byte and gas here + how did we get these numbers?

**Review by PaddyMc — APPROVED** [review]

File: app/upgrades/v22/upgrades.go (hunk 3 above)
Comment (czarcas7ic) [reply app/upgrades/v22/upgrades.go:32]: We utilize all defaults, except for the two block params that the in line comment links to.

**Comment by ValarDragon** [comment]
We should definitely add a changelog for this

File: app/upgrades/v22/upgrades.go (hunk 3 above)
Comment (ValarDragon) [reply app/upgrades/v22/upgrades.go:32]: Can the code comment paste what mainnet used to have>

Comment (czarcas7ic) [reply app/upgrades/v22/upgrades.go]: Added previous mainnet values in-line comments here https://github.com/osmosis-labs/osmosis/pull/7220/commits/cb054741ed023818ddc8e0d98a3333a293c0b998

**Comment by czarcas7ic** [comment]
@ValarDragon there is a changelog entry https://github.com/osmosis-labs/osmosis/blob/cb054741ed023818ddc8e0d98a3333a293c0b998/CHANGELOG.md?plain=1#L59

**Review by ValarDragon — APPROVED** [review]

**Pull request merged**